internal/app/machined/pkg/runtime/v1alpha1/platform/container/container.go

```diff
@@ -77,6 +78,34 @@ func (c *Container) NetworkConfiguration(ctx context.Context, _ state.State, ch
 
 	networkConfig.Hostnames = append(networkConfig.Hostnames, hostnameSpec)
 
+	resolverSpec := network.ResolverSpecSpec{
+		ConfigLayer: network.ConfigPlatform,
+	}
+
+	resolvers, err := os.ReadFile("/etc/resolv.conf")
+	if err != nil {
+		return err
+	}
+
+	for _, line := range bytes.Split(resolvers, []byte("\n")) {
```

Comment:
Maybe use `bufio.NewScanner(f)` and `f, err := os.Open("/etc/resolv.conf")`? 

That way there is no need for `[]byte(...)` conversions.

Reply:
let's call it PoC for now, I think this way is actually easier for a small file.